pylatex/tikz.py

```diff
@@ -34,13 +46,466 @@ def __init__(self, options=None, *, data=None):
         super().__init__(options=options, data=data)
 
 
+class TikZScope(Environment):
+    """TikZ Scope Environment."""
+
+    _latex_name = 'scope'
+
+
+class TikZCoordinate(object):
+    """A General Purpose Coordinate Class."""
+
+    _coordinate_str_regex = re.compile(r'(\+\+)?\(\s*(-?[0-9]+(\.[0-9]+)?)\s*'
+                                       r',\s*(-?[0-9]+(\.[0-9]+)?)\s*\)')
+
+    def __init__(self, x, y, relative=False):
+        """
+        Args
+        ----
+        x: float or int
+            X coordinate
+        y: float or int
+            Y coordinate
+        relative: bool
+            Coordinate is relative or absolute
+        """
+        self._x = float(x)
+        self._y = float(y)
+        self.relative = relative
+
+    def __repr__(self):
+        if self.relative:
+            ret_str = '++'
+        else:
+            ret_str = ''
+        return ret_str + '({},{})'.format(self._x, self._y)
+
+    def dumps(self):
+        """Return representation."""
+
+        return self.__repr__()
+
+    @classmethod
+    def from_str(cls, coordinate):
+        """Build a TikZCoordinate object from a string."""
+
+        m = cls._coordinate_str_regex.match(coordinate)
+
+        if m is None:
+            raise ValueError('invalid coordinate string')
+
+        if m.group(1) == '++':
+            relative = True
+        else:
+            relative = False
+
+        return TikZCoordinate(
+            float(m.group(2)), float(m.group(4)), relative=relative)
+
+    def __eq__(self, other):
+        if isinstance(other, tuple):
```

Comment:
This seems like it can be simplified by creating an other_x and other_y variable. Then the rest of the code can be shared between both comparisons. Also you can change the if to just return the check itself instead of returning true when the check is true and false when the check is false.